components/user-mgt/org.wso2.carbon.identity.user.account.association/src/main/java/org/wso2/carbon/identity/user/account/association/UserAccountAssociationService.java

```diff
@@ -58,12 +243,153 @@ public void createUserAccountAssociation(String userName, char [] password) thro
+        try {
+            if (!UserAccountAssociationDAO.getInstance().isValidUserAssociation(domainName, tenantId,
+                    tenantAwareUsername)) {
+                if (log.isDebugEnabled()) {
+                    log.debug(UserAccountAssociationConstants.ErrorMessages.INVALID_ASSOCIATION.getDescription());
+
+                }
+                throw new UserAccountAssociationClientException(UserAccountAssociationConstants
+                        .ErrorMessages.INVALID_ASSOCIATION.toString());
+            }
+            UserAccountConnectorImpl.getInstance().deleteUserAccountAssociation(userName);
+        } catch (UserAccountAssociationServerException e) {
+            log.error(UserAccountAssociationConstants.ErrorMessages.CONN_DELETE_ERROR.getDescription(), e);
+            throw new UserAccountAssociationClientException(UserAccountAssociationConstants.ErrorMessages
+                    .CONN_DELETE_ERROR.toString());
+        } catch (UserAccountAssociationException e) {
+            throw (UserAccountAssociationClientException) e;
+        }
+    }
+
+    /**
+     * Delete account association of logged in user
+     *
+     * @throws org.wso2.carbon.identity.user.account.association.exception.UserAccountAssociationClientException
+     */
+    public void deleteMyAssociations() throws UserAccountAssociationClientException {
+
+        String userName = UserCoreUtil.addTenantDomainToEntry(CarbonContext.getThreadLocalCarbonContext()
+                .getUsername(), CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+        try {
+            UserAccountConnectorImpl.getInstance().deleteUserAccountAssociation(userName);
+        } catch (UserAccountAssociationServerException e) {
+            log.error(UserAccountAssociationConstants.ErrorMessages.CONN_DELETE_ERROR.getDescription(), e);
+            throw new UserAccountAssociationClientException(UserAccountAssociationConstants.ErrorMessages
+                    .CONN_DELETE_ERROR.toString());
+        } catch (UserAccountAssociationException e) {
+            throw (UserAccountAssociationClientException) e;
+        }
+    }
+
+    /**
+     * Delete an existing user account association as admin
+     *
+     * @param userName
+     * @throws org.wso2.carbon.identity.user.account.association.exception.UserAccountAssociationClientException
+     */
+    public void deleteAssociationsOfUser(String userName) throws UserAccountAssociationClientException {
+
+        try {
+            UserAccountConnectorImpl.getInstance().deleteUserAccountAssociation(userName);
+        } catch (UserAccountAssociationServerException e) {
+            log.error(UserAccountAssociationConstants.ErrorMessages.CONN_DELETE_ERROR.getDescription(), e);
+            throw new UserAccountAssociationClientException(UserAccountAssociationConstants.ErrorMessages
+                    .CONN_DELETE_ERROR.toString());
```

Comment:
This is a admin service, here I have logged server exception without sending it to the client side and throw a client exception with user friendly error code and message.